controllers/host/host_controller.go

```diff
@@ -1383,6 +1385,42 @@ func (r *HostReconciler) ReconcileExistingHost(client *gophercloud.ServiceClient
 
 	logHost.Info("current config is:", "values", current)
 
+	specBytes, err := json.Marshal(profile)
+	if err != nil {
+		logHost.Info("Failed to marshal profile data: %s", err)
+	}
+
+	currentBytes, err := json.Marshal(current)
+	if err != nil {
+		logHost.Info("Failed to marshal current data: %s", err)
+	}
+
+	var specData map[string]interface{}
+	var currentData map[string]interface{}
+
+	err = json.Unmarshal([]byte(specBytes), &specData)
+	if err != nil {
+		logHost.Info("Failed to parse profile data: %s", err)
+	}
+
+	err = json.Unmarshal([]byte(currentBytes), &currentData)
+	if err != nil {
+		logHost.Info("Failed to parse current data: %s", err)
+	}
+
+	diff := cmp.Diff(currentData, specData)
+	deltaString := common.CollectDiffValues(diff, common.HostProperties)
```

Comment:
Not mandatory but personally, I prefer to extract these marshal/unmarshal lines to the new function.

Reply:
You are right, I've moved this entire block to a new function, it's clearer now. 
Thanks Tak!